From 3ef4b68659614d5117a8555d39eb35b9afbd8b15 Mon Sep 17 00:00:00 2001
From: Jeffy Chen <jeffy.chen@rock-chips.com>
Date: Wed, 5 Jan 2022 14:56:17 +0800
Subject: [PATCH 01/93] Revert "compositor: avoid setting
 WAYLAND_DISPLAY=wayland-0"

This reverts commit ae69381b6eb43ab63992906d1bf29ffd2b4df10d.

Use "wayland-0" by default.

Signed-off-by: Jeffy Chen <jeffy.chen@rock-chips.com>
---
 compositor/main.c | 23 +++++++++--------------
 1 file changed, 9 insertions(+), 14 deletions(-)

diff --git a/compositor/main.c b/compositor/main.c
index 15f9d4e..631355e 100644
--- a/compositor/main.c
+++ b/compositor/main.c
@@ -858,29 +858,24 @@ handle_primary_client_destroyed(struct wl_listener *listener, void *data)
 static int
 weston_create_listening_socket(struct wl_display *display, const char *socket_name)
 {
-	char name_candidate[32];
-
 	if (socket_name) {
 		if (wl_display_add_socket(display, socket_name)) {
 			weston_log("fatal: failed to add socket: %s\n",
 				   strerror(errno));
 			return -1;
 		}
-
-		setenv("WAYLAND_DISPLAY", socket_name, 1);
-		return 0;
 	} else {
-		for (int i = 1; i <= 32; i++) {
-			sprintf(name_candidate, "wayland-%d", i);
-			if (wl_display_add_socket(display, name_candidate) >= 0) {
-				setenv("WAYLAND_DISPLAY", name_candidate, 1);
-				return 0;
-			}
+		socket_name = wl_display_add_socket_auto(display);
+		if (!socket_name) {
+			weston_log("fatal: failed to add socket: %s\n",
+				   strerror(errno));
+			return -1;
 		}
-		weston_log("fatal: failed to add socket: %s\n",
-			   strerror(errno));
-		return -1;
 	}
+
+	setenv("WAYLAND_DISPLAY", socket_name, 1);
+
+	return 0;
 }
 
 WL_EXPORT void *
-- 
2.20.1

